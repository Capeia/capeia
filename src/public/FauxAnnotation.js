/**
 * The FauxAnnotation is an annotation, similar to the annotations found in articles,
 * that can be used as a plain component.
 *
 * TODO: At some point this should be unified with the article annotation component.
 * @flow
 */

import React from 'react'
import { Popover, OverlayTrigger } from 'react-bootstrap'
import withStyles from 'isomorphic-style-loader/lib/withStyles'
import sd from './shared/rich-text-editor/entity/Decorator.scss'
import s from './FauxCitation.scss'
import annotationIcon from 'article/glyphicon-pushpin.png'

const IDEOGRAPHIC_SPACE = '\u3000'

type Props = {
  children?: $FlowFixMe,
  id: string
}

class FauxAnnotation extends React.Component {
  props: Props

  renderOverlay = () =>
    <Popover id={this.props.id}>{this.props.children}</Popover>

  render () {
    return (
      <OverlayTrigger trigger='click' placement='top' overlay={this.renderOverlay()}>
        <span className={`${sd.ImageDecorator} ${s.icon}`} style={{backgroundImage: `url(${annotationIcon})`}}>
          {IDEOGRAPHIC_SPACE}
        </span>
      </OverlayTrigger>
    )
  }
}

export default withStyles(s, sd)(FauxAnnotation)
