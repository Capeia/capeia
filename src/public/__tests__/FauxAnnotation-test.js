// @flow
/* eslint-env jest, jasmine */

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import FauxAnnotation from '../FauxAnnotation'

describe('<FauxAnnotation>', () => {
  it('renders correctly', () => {
    const Component = require('../FauxAnnotation').default
    const tree = renderer.create(
      <Component id='123'>Foo Bar</Component>
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render a popover containing children', () => {
    const text = <span>foo bar baz</span>
    const wrapper = shallow(<FauxAnnotation id='123'>{text}</FauxAnnotation>)
    const trigger = wrapper.find('OverlayTrigger')
    expect(trigger.length).toEqual(1)
    const popover = trigger.props().overlay
    expect(popover.props.id).toEqual('123')
    expect(popover.props.children).toEqual(text)
  })
})
