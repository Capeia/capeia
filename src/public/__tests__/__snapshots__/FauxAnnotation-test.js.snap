// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FauxAnnotation> renders correctly 1`] = `
<span
  className="ImageDecorator icon"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(article/glyphicon-pushpin.png)",
    }
  }
>
  ã€€
</span>
`;
